test/functional/local_transaction_controller_test.rb

```diff
@@ -186,56 +180,35 @@ def subscribe_logstasher_to_postcode_error_notification
       setup do
         stub_mapit_does_not_have_a_bad_postcode("BLAH")
 
-        subscribe_logstasher_to_postcode_error_notification
-
         post :search, params: { slug: "send-a-bear-to-your-local-council", postcode: "BLAH" }
       end
 
-      should "expose the 'invalid postcode format' error to the view" do
-        location_error = assigns(:location_error)
-        assert_equal location_error.postcode_error, "invalidPostcodeFormat"
-      end
-
-      should "log the 'invalid postcode format' error to the view" do
-        assert_equal(LogStasher.store["postcode_error_notification"], postcode_error: "invalidPostcodeFormat")
+      should "expose the 'invalid postcode format' error in the view" do
+        assert_match "invalidPostcodeFormat", response.body
       end
     end
 
     context "loading the local transaction when posting a postcode with no matching areas" do
       setup do
         stub_mapit_does_not_have_a_postcode("WC1E 9ZZ")
 
-        subscribe_logstasher_to_postcode_error_notification
-
         post :search, params: { slug: "send-a-bear-to-your-local-council", postcode: "WC1E 9ZZ" }
       end
 
-      should "expose the 'no mapit match' error to the view" do
-        location_error = assigns(:location_error)
-        assert_equal location_error.postcode_error, "fullPostcodeNoMapitMatch"
-      end
-
-      should "log the 'no mapit match' error to the view" do
-        assert_equal(LogStasher.store["postcode_error_notification"], postcode_error: "fullPostcodeNoMapitMatch")
+      should "expose the 'no mapit match' error in the view" do
+        assert_match "fullPostcodeNoMapitMatch", response.body
       end
     end
 
     context "loading the local transaction when posting a location that has no matching local authority" do
       setup do
         stub_mapit_has_a_postcode_and_areas("AB1 2CD", [0, 0], [])
 
-        subscribe_logstasher_to_postcode_error_notification
-
         post :search, params: { slug: "send-a-bear-to-your-local-council", postcode: "AB1 2CD" }
       end
 
-      should "expose the 'missing local authority' error to the view" do
-        location_error = assigns(:location_error)
-        assert_equal "noLaMatch", location_error.postcode_error
-      end
-
-      should "log the 'missing local authority' error to the view" do
-        assert_equal(LogStasher.store["postcode_error_notification"], postcode_error: "noLaMatch")
+      should "expose the 'missing local authority' error in the view" do
+        assert_match "noLaMatch", response.body
```

Comment:
I think this is also passing because `noLaMatch` is part of the analytics actions in `response.body`. The error on the view would be something like this:

```
assert_match "We couldn&#39;t find a council for this postcode", response.body
```

I tried to display with `I18n.t` but didn't seem to work...

Reply:
Ah I did this to maintain it testing the code aspect since it's used for analytics. On reflection I think your suggestion is better and my analytics tests are rather opaque.

Reply:
👍